### PR TITLE
Handle an env var that is set to the empty string

### PR DESCRIPTION
**Pull request by jennybc**
Even if we don't plan to use this approach in, e.g., tests, we need to not error on this edge case, at the very least

### CODE REVIEW

File: tests/testthat/test.R [hunk 1]
```diff
@@ -1,8 +1,8 @@
+test_that("gitcreds handles env var set to empty string", {
+  orig <- Sys.getenv("GITHUB_PAT_GITHUB_ACME_COM")
+  on.exit(Sys.setenv(GITHUB_PAT_GITHUB_ACME_COM = orig), add = TRUE)
```
Comment (jennybc) [inline tests/testthat/test.R:3]: I would much prefer to use withr, due to the fiddliness of setting/resetting env vars, because this doesn't really handle all scenarios on all OSes.

Comment (gaborcsardi) [reply tests/testthat/test.R:3]: you can do it like this: https://github.com/r-lib/gitcreds/blob/0ebc53fde7d2dd5c3875c666ed352756cbaac988/R/git-auth.R#L653-L654

**Comment by gaborcsardi** [comment]
I am afraid that this will not work because on Windows you cannot set an env var to an empty value. This is why the tests are failing on Windows.

**Comment by jennybc** [comment]
> I am afraid that this will not work because on Windows you cannot set an env var to an empty value. This is why the tests are failing on Windows.

Well, we could skip this test on Windows.

The real points to settle are:

1. How to handle the empty string case or anything that could possibly lead to `length(spval) == 0`
1. How to make it easy for us to explore and test the scenario of "user has no matching token in the store". Here or in downstream packages. Because there has to be a better way than me constantly deleting my PAT from my keychain.

**Comment by gaborcsardi** [comment]
> 1. How to handle the empty string case or anything that could possibly lead to length(spval) == 0

Same as when it is not set I think?

> 2. How to make it easy for us to explore and test the scenario of "user has no matching token in the store". Here or in downstream packages. Because there has to be a better way than me constantly deleting my PAT from my keychain.

Use a (random) host name that surely has no PAT?

**Comment by jennybc** [comment]
>> How to handle the empty string case or anything that could possibly lead to length(spval) == 0
> Same as when it is not set I think?

So see if #11 is what you want.

**Comment by gaborcsardi** [comment]
Since #11 was merged, I think we don't need this any more.
